python/orca/test/bigdl/orca/learn/ray/tf/test_tf2estimator_ray_dataset.py

```diff
@@ -0,0 +1,120 @@
+# limitations under the License.
+#
+import pytest
+from unittest import TestCase
+
+import ray
+from ray.data import Dataset
+import tensorflow as tf
+import numpy as np
+
+from bigdl.orca.learn.tf2 import Estimator
+from bigdl.orca import init_orca_context, stop_orca_context
+
+def train_data_creator(a=5, b=10, size=1000):
+    def get_dataset(a, b, size) -> Dataset:
+        items = [i / size for i in range(size)]
+        dataset = ray.data.from_items([{
+            "x": x,
+            "y": a * x + b
+        } for x in items])
+        return dataset
+    train_dataset = get_dataset(a, b, size)
+    return train_dataset
+
+def val_data_creator(a=5, b=10, size=100):
+    def get_dataset(a, b, size) -> Dataset:
+        items = [i / size for i in range(size)]
+        dataset = ray.data.from_items([{
+            "x": x,
+            "y": a * x + b
+        } for x in items])
+        return dataset
+    val_dataset = get_dataset(a, b, size)
+    return val_dataset
+
+def simple_model(config):
+    model = tf.keras.Sequential([
+            tf.keras.layers.InputLayer(input_shape=(1, )),
+            tf.keras.layers.Dense(10),
+            tf.keras.layers.Dense(1)])
+    return model
+
+def compile_args(config):
+    if config is None:
+        lr = 1e-3
+    else:
+        lr = config["lr"]
+    args = {
+        "optimizer": tf.keras.optimizers.SGD(lr),
+        "loss": "mean_squared_error",
+        "metrics": ["mean_squared_error"]}
+    return args
+
+def model_creator(config):
+    model = simple_model(config)
+    model.compile(**compile_args(config))
+    return model
+
+def get_estimator(workers_per_node=2, model_fn=model_creator):
+    estimator = Estimator.from_keras(model_creator=model_fn,
+                                     config={"lr": 1e-3},
+                                     workers_per_node=workers_per_node,
+                                     backend="tf2")
+    return estimator
+
+class TestTF2Estimator(TestCase):
+    def setUp(self):
+        init_orca_context(runtime="ray", address="localhost:6379")
+
+    def tearDown(self):
+        stop_orca_context()
+
+    def test_train_and_evaluate(self):
+        orca_estimator = get_estimator(workers_per_node=2)
+        train_dataset = train_data_creator()
+        validation_dataset = val_data_creator()
+
+        data_config_args = {"output_signature":
+                            (tf.TensorSpec(shape=(None, 1), dtype=tf.float32),
+                             tf.TensorSpec(shape=(None), dtype=tf.float32))
+                            }
+        
+        start_eval_stats = orca_estimator.evaluate(data=validation_dataset,
+                                                   num_steps=2, batch_size=32,
+                                                   label_cols="x", feature_cols=["y"],
+                                                   data_config=data_config_args)
+        print(start_eval_stats)
```

Comment:
maybe we should check whether the stats are as expected instead of just printing them out, so that the test could fail if anything goes wrong.